### DISCUSSION TITLE
Scope and Direction

### DISCUSSION THREAD

**Discussion by huangziwei** [jaxleyverse/jaxley-mech#3] [General]
This thread is about the scope and directions of `jaxley-mech`.

`jaxley-mech` will be built on top of `jaxley` and offer wide-range of paper-specific implementations of ion channels, synapses and receptors. 

That said, there's actually no strong reasons against simply adding them to `jaxley` directly. But my preference right now is to keep `jaxley` as an inference machine (with a few canonical examples of how to implement channels, synapses and receptors there), while offers paper-specific implementations in `jaxley-mech`.

UPDATE (20230130)

I think I will prioritize going through the NeuronC documentation and porting the relevant parts, probably for quite a while

**Comment by philippberens** [comment]
This sounds good to me. Why do you focus on papers as the organizing unit?

I think it would be nice to provide also a template implementation e.g. of channels we recommend when simulation mouse retina

**Reply by huangziwei** [reply]
the main rationale is that the rate constants for the gating variables vary from paper to paper. They are also just some parameters fitted by the experimental data (such as voltage clamp data) to determine a channel's kinetics, but in most papers they are presented as fixed constants.

**Reply by huangziwei** [reply]
maybe after implementing enough mechanisms, we can add some presets:

```python
from jaxley_mech.presets import retina_mouse, retina_cat # etc 
``` 

it would also be nice to be able to generate a table of all mechanisms in the presets with reference.

**Comment by philippberens** [comment]
Ok, makes sense... do we have a strategy for synapses yet?

**Reply by huangziwei** [reply]
no. need to read some papers first.

**Reply by huangziwei** [reply]
a few receptors (AMPA, NMDA, GABAa/b) are now implemented as synapses:

https://github.com/jaxleyverse/jaxley-mech/blob/main/notebooks/synapses/dms98.ipynb

**Comment by michaeldeistler** [comment]
Sorry for the late reply, I had not activated notifications for this repo...but yes, keeping this as a separate repo sounds good to me!
